// [[Reason: Implicit undefined in parameter can only be detected by TSC.]] ////

//// [tests/cases/compiler/isolatedDeclarationsAddUndefined.ts] ////

===================================================================
--- TSC	declarations
+++ DTE	declarations
@@ -10,18 +10,17 @@
 }
 export {};
 
 //// [file2.d.ts]
-export declare function foo(p?: (ip: invalid, v: number) => void): void;
+export declare function foo(p?: (ip: number, v: number) => void): void;
 export declare class Bar2 {
     readonly r = 1;
     f: number;
 }
 
 /// [Errors] ////
 
 file1.ts(4,11): error TS9012: Property must have an explicit type annotation with --isolatedDeclarations.
-file2.ts(1,26): error TS9026: Declaration emit for this parameter requires implicitly adding undefined to it's type. This is not supported with --isolatedDeclarations.
 
 
 ==== file1.ts (1 errors) ====
     type N = 1;
@@ -34,13 +33,10 @@
         readonly r = 1;
         f = 2;
     }
     
-==== file2.ts (1 errors) ====
+==== file2.ts (0 errors) ====
     export function foo(p = (ip = 10, v: number): void => {}): void{
-                             ~~~~~~~
-!!! error TS9026: Declaration emit for this parameter requires implicitly adding undefined to it's type. This is not supported with --isolatedDeclarations.
-!!! related TS9029 file2.ts:1:26: Add a type annotation to the parameter ip.
     }
     export class Bar2 {
         readonly r = 1;
         f = 2;
